### PR TITLE
SIR-2289 Add ability to read and substitute values with secrets in json files

### PR DESCRIPTION
**Pull request by avolkov-dev**
Logging configuration without secrets
image

### CODE REVIEW

File: src/Swisschain.Sdk.Server/Common/HostOptionsBuilder.cs [hunk 1]
```diff
@@ -21,6 +22,8 @@ public HostOptionsBuilder()
         internal int GrpcPort { get; set; }
 
         internal WebJsonConfigurationSourcesBuilder WebJsonConfigurationSourcesBuilder { get; }
+        
+        internal FileJsonSettingsLocations FileJsonSettingsLocations { get; private set; }
```
Comment (KonstantinRyazantsev) [inline src/Swisschain.Sdk.Server/Common/HostOptionsBuilder.cs:26]: `FileJsonConfigurationLocation` to fit approach two lines above

File: src/Swisschain.Sdk.Server/Common/HostOptionsBuilder.cs [hunk 2]
```diff
@@ -64,5 +67,11 @@ public HostOptionsBuilder AddWebJsonConfigurationSources(IReadOnlyCollection<str
 
             return this;
         }
+
+        public HostOptionsBuilder AddFileJsonSettingsLocations(FileJsonSettingsLocations locations)
```
Comment (KonstantinRyazantsev) [inline src/Swisschain.Sdk.Server/Common/HostOptionsBuilder.cs:71]: AddFileJsonConfigLocation

File: src/Swisschain.Sdk.Server/Configuration/FileJsonSettingsLocations.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+﻿using System.Linq;
+using Microsoft.Extensions.Configuration;
+
+namespace Swisschain.Sdk.Server.Configuration
+{
+    public class FileJsonSettingsLocations
```
Comment (KonstantinRyazantsev) [inline src/Swisschain.Sdk.Server/Configuration/FileJsonSettingsLocations.cs:6]: FileJsonConfigLocation

File: src/Swisschain.Sdk.Server/Configuration/FileJsonSettingsLocations.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
+﻿using System.Linq;
+using Microsoft.Extensions.Configuration;
+
+namespace Swisschain.Sdk.Server.Configuration
+{
+    public class FileJsonSettingsLocations
+    {
+        public FileJsonSettingsLocations()
+        {
+            SecretsFilePath = new string[0];
```
Comment (KonstantinRyazantsev) [inline src/Swisschain.Sdk.Server/Configuration/FileJsonSettingsLocations.cs:10]: Array.Empty<string>()

File: src/Swisschain.Sdk.Server/Configuration/FileJsonSettingsLocations.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+            new ConfigurationBuilder()
+                .AddJsonFile("settings/settingslocator.json", optional: true)
+                .AddEnvironmentVariables()
+                .Build()
+                .Bind(result);
+
+            return result;
+        }
+        
+        public string[] SecretsFilePath { get; set; }
```
Comment (KonstantinRyazantsev) [inline src/Swisschain.Sdk.Server/Configuration/FileJsonSettingsLocations.cs:27]: .

Comment (KonstantinRyazantsev) [reply src/Swisschain.Sdk.Server/Configuration/FileJsonSettingsLocations.cs:27]: SecretFilePaths

File: src/Swisschain.Sdk.Server/Configuration/FileJsonSettingsLocations.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        public string[] SettingsFilePath { get; set; }
```
Comment (KonstantinRyazantsev) [inline src/Swisschain.Sdk.Server/Configuration/FileJsonSettingsLocations.cs:28]: SettingFilePaths

File: src/Swisschain.Sdk.Server/Configuration/FileJsonSettingsLocations.cs (hunk 3 above)
Comment (KonstantinRyazantsev) [reply src/Swisschain.Sdk.Server/Configuration/FileJsonSettingsLocations.cs:6]: Let's make this class immutable and add a builder for it

File: src/Swisschain.Sdk.Server/Common/HostOptionsBuilder.cs (hunk 2 above)
Comment (KonstantinRyazantsev) [reply src/Swisschain.Sdk.Server/Common/HostOptionsBuilder.cs:71]: Let's use Func<FileJsonConfigLocationBuilder> optionsBuilder as the parameter here

File: src/Swisschain.Sdk.Server/Configuration/FileJsonSettingsLocations.cs (hunk 3 above)
Comment (avolkov-dev) [reply src/Swisschain.Sdk.Server/Configuration/FileJsonSettingsLocations.cs:6]: I guess It's not possible cause we are binding this class from configuration.

**Review by KonstantinRyazantsev — APPROVED** [review]

**Review by KonstantinRyazantsev — APPROVED** [review]

**Pull request merged**